code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -501,19 +499,20 @@ GLOBAL_LIST_EMPTY(scan_typecaches)
 	if(!scan_cache)
 		scan_cache = typecacheof(scan_types)
 		GLOB.scan_typecaches[key] = scan_cache
-	var/turf/current_turf = get_turf(src)
-	if(!current_turf)
+	if(!get_turf(src))
 		return
-	var/list/adjacent = current_turf.get_atmos_adjacent_turfs(1)
-	if(shuffle_adjacent) //If we were on the same tile as another bot, let's randomize our choices so we dont both go the same way
-		adjacent = shuffle(adjacent)
-		shuffle_adjacent = FALSE
+	// Nicer behavior, ensures we don't conflict with other bots quite so often
+	var/list/adjacent = list()
+	for(var/turf/to_walk in view(1, src))
```

Comment:
why not block() if you only want turfs

Reply:
because I only want turfs in view

Reply:
adding as turf in view() speeds it up i thi k cuz of compiler shit

Reply:
I don't want to expose the children to as(), but that is tempting